flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobStatusObserver.java

```diff
@@ -72,22 +81,92 @@ public boolean observe(
         }
 
         if (!clusterJobStatuses.isEmpty()) {
+            // There are jobs on the cluster, we filter the ones for this resource
             Optional<JobStatusMessage> targetJobStatusMessage =
                     filterTargetJob(jobStatus, clusterJobStatuses);
+
             if (targetJobStatusMessage.isEmpty()) {
-                jobStatus.setState(org.apache.flink.api.common.JobStatus.RECONCILING.name());
+                LOG.warn("No matching jobs found on the cluster");
+                ifRunningMoveToReconciling(jobStatus, previousJobStatus);
+                // We could list the jobs but cannot find the one for this resource
+                if (resource instanceof FlinkDeployment) {
+                    // This should never happen for application clusters, there is something wrong
+                    setUnknownJobError((FlinkDeployment) resource);
```

Comment:
Could this be normal if we concurrently removed a job as well with a complex configuration change?

Reply:
Not really, this means that there was a job on the cluster that we cannot match to the application deployment. this should never happen.